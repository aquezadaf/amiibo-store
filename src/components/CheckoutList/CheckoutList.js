import React from 'react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next';
import styles from './CheckoutList.module.css'

function CheckoutList(props) {
    const { products } = props
    const { t } = useTranslation()
    return (
        <div className={styles.container}>
            <div className={styles.title}>{t('checkoutList.title')}</div>
            <div className={styles.products}>
                {products.map(product => <div key={product.id}>
                    <div className={styles.product}>
                        <img className={styles.image} src={product.image} alt="amiibo" />
                        <div>
                            <div className={styles.name}>{product.name}</div>
                            <div className={styles.price}>$ {product.price} x {product.quantity}</div>
                        </div>
                    </div>
                </div>)}
                <div className={styles.totalContainer}>
                    <span className={styles.total}>{t('checkoutList.total')}</span>
                    <span className={styles.total}>${products.reduce((total, { quantity, price }) => total + (quantity * price), 0)}</span>
                </div>
            </div>
        </div>
    )
}

CheckoutList.propTypes = {
    products: PropTypes.arrayOf(PropTypes.shape({
        image: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
        quantity: PropTypes.number.isRequired
    }))
}

export default CheckoutList
