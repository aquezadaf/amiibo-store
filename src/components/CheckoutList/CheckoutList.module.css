.container {
    margin: 10px;
}

.title {
    margin-left: 10px;
    margin-bottom: 20px;
    font-size: 30px;
    font-weight: bold;
    text-align: left;
}

.products {
    display: flex;
    flex-direction: column;
}

.product {
    display: flex;
    background-color: #fafafa;
    padding: 15px;
    margin: 10px 0;
    border-radius: 15px;
    justify-content: space-between;
    align-items: center;
}

.image {
    display: block;
    max-width:80px;
    max-height:100px;
}

.name {
    font-size: 18px;
    color: #393c45;
    font-weight: 700;
    text-align: right;
    padding-bottom: 10px;
}

.price {
    color: #393c45;
    font-size: 18px;
    text-align: right;
    padding-bottom: 8px;
}

.totalContainer {
    display: flex;
    justify-content: space-between;
    padding-top: 20px;
}

.total {
    font-size: 25px;
    color: #393c45;
    font-weight: 700;
}
