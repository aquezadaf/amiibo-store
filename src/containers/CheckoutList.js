import { connect } from 'react-redux'
import CheckoutList from '../components/CheckoutList'
import selectors from '../selectors'

const mapStateToProps = state => ({
    products: selectors.shoppingCart.productsSelector(state)
});

export default connect(mapStateToProps, null)(CheckoutList);
