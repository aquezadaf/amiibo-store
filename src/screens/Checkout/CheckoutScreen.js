import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import CheckoutList from "../../components/CheckoutList";
import styles from "./CheckoutScreen.module.css";

function CheckoutScreen(props) {
  const { products } = props;
  const { t } = useTranslation();
  if (!products.length) {
    return (
      <div className={styles.container}>
        <div className={styles.emptyMessage}>{t("checkout.empty")}</div>
      </div>
    );
  }
  return (
    <div className={styles.container}>
      <CheckoutList products={products} />
    </div>
  );
}

CheckoutScreen.propTypes = {
  products: PropTypes.array.isRequired,
};

export default CheckoutScreen;
