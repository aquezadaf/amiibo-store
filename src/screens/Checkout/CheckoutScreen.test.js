import React from 'react'
import { shallow } from 'enzyme'
import CheckoutScreen from './CheckoutScreen'
import CheckoutList from "../../components/CheckoutList";

jest.mock("../../components/CheckoutList", () => () => null)

it('renders message when receiving empty products', () => {
    const wrapper = shallow(<CheckoutScreen products={[]} />)
    expect(wrapper.html()).toMatchSnapshot()
})

it('renders CheckoutList with products when at least one product', () => {
    const wrapper = shallow(<CheckoutScreen products={[{ id: 1 }]} />)
    const { products } = wrapper.find(CheckoutList).props()
    expect(products).toEqual([{ id: 1 }])
})
