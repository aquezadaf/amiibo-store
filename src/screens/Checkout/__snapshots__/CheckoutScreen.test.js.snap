// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders message when receiving empty products 1`] = `"<div class=\\"container\\"><div class=\\"emptyMessage\\">checkout.empty</div></div>"`;
